system/jlib/jlog.cpp

```diff
@@ -1678,7 +1679,37 @@ void CLogMsgManager::resetMonitors()
     suspendChildren();
     removeAllMonitors();
     Owned<ILogMsgFilter> defaultFilter = getDefaultLogMsgFilter();
-    addMonitor(theStderrHandler, defaultFilter);
+
+    // dont add log handler if stderr is redirected to /dev/null ...
+
+    bool okToAddMonitor = true;
+#ifndef _WIN32
+    int fd2 = fileno(stderr);
+    if (fd2 >= 0)
```

Comment:
I would extract this into a separate function isRedirectedToDevNull() or something similar.  (Including the windows #ifdef).

Reply:
ok, sure.  You mean like this ?  (static in same file) or something external in jfile etc. that takes a FILE * arg ?
When it looks ok I will squash and create a JIRA etc.